speedtest/client.go

```diff
@@ -13,7 +13,20 @@ import (
 	"sync"
 )
 
-type Client struct {
+type Client interface {
```

Comment:
Hi @sunol, this is for the unit test purpose, so that we can exchange the Client instance to fake the underlining behavior.  I've listed all the external methods, so that behavior of Client instance methods doesn't change.  I don't believe you want to expose all those methods but I've listed all the external methods, for the backward compatibility purpose.

I added this so that we can have a unit test to reproduce the crash issue.  We can separate the PR between the fix and the unit test, but I believe it's better to have a single PR which contains both fix and the test for the future reference purpose.

Let me know what you think, @surol.